### PR TITLE
PCF-480 Implement the subtests table component

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**

### CODE REVIEW

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [inline src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:104]: Can be moved to platform.ts -> platform.tsx

```
import { type ReactNode } from 'react';
import AppleIcon from '@mui/icons-material/Apple';
import AndroidIcon from '../components/Shared/Icons/AndroidIcon';
import LinuxIcon from '../components/Shared/Icons/LinuxIcon';
import WindowsIcon from '../components/Shared/Icons/WindowsIcon';


export const platformIcons: Record<PlatformShortName, ReactNode> = {
  Linux: <LinuxIcon />,
  OSX: <AppleIcon />,
  Windows: <WindowsIcon />,
  Android: <AndroidIcon />,
  Unspecified: '',
};
```

File: src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx
Comment (beatrice-acasandrei) [inline src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx:125]:
```suggestion
    result.test.includes(searchTerm)
```

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [inline src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:71]:
```suggestion
function getSuite(
```

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [inline src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:94]: Move function to `src/utils/helpers.ts`.

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (beatrice-acasandrei) [inline src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:25]: I have a typescript error here.

File: src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx:125]: Since all the other fields are the same for each result except the test name then I suggest excluding them from the search functionality.

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (Carla-Moz) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:24]: For the typescript error, it might get fixed if you get `results` data directly in `SubtestsRevisionHeader` instead. This assumes you've deferred the returned data from the loaders as mentioned in my comment below. 

See here: [Await component docs](https://reactrouter.com/en/main/components/await) and here: [UseAsyncValue docs](https://reactrouter.com/en/main/hooks/use-async-value)

A suggestion:


```
import { useAsyncValue } from 'react-router-dom';
....

function SubtestsRevisionHeader() {
  const loaderData = useAsyncValue();
  const resultsData = loaderData as CompareResultsItem[][];

 //filter results if needed

  const results = filteredResults as CompareResults[];


  const subtestsHeader: SubtestsRevisionsHeader = {
    suite: results[0].suite,
    framework_id: results[0].framework_id,
    test: results[0].test,
    option_name: results[0].option_name,
    extra_options: results[0].extra_options,
    new_rev: results[0].new_rev,
    new_repo: results[0].new_repository_name,
    platform: results[0].platform,
  };

....
```

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (Carla-Moz) [inline src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:64]: Results are resolved here. Could explain why you have the typescript error. So you might have to create the header within the children of `Await`.

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (Carla-Moz) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:102]: Honestly, I'd create `platform.tsx` in the `Results` folder. See what others think. I agree it would be confusing to have another `.tsx` file in the utils folder.

**Review by Carla-Moz — COMMENTED** [review]
Looks great! I left some comments that I think might resolve some of your typescript errors.

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:102]: @julienw What do you think?

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (julienw) [inline src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:52]: to be removed?

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (julienw) [inline src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:53]: at this point it could be better to put the result value rather than calculate it

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (julienw) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:57]: yeah I agree, let's keep it simple for now.

File: src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx
Comment (julienw) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx:196]: As a bit of nuance, typestyle is costly when it's used at render time -- I think its normal use should be to be called outside of components, and then the performance is OK: the style is registered once at load time, the result is then only a class name so it's pretty cheap.

But otherwise I agree; we probably don't need typestyle in general, but more because it adds yet another way of styling. Especially we should look at using `styled` more (integrated in mui) and also use MUI's theming mechanism better.

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:52]: This is something I left the same as in the RevisionHeader component

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:57]: I removed `Suspense` and added a new clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488), we'll consider there adding a spinner on initial load.

File: src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx:196]: I addressed it, I'll leave the comment open for review purposes.

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:53]: I also added this suggestion in the clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488) for ResvisionHeader

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionHeader.tsx:102]: I added this suggestion in the clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488)

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionRow.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionRow.tsx:23]: I added this suggestion in the clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488) and it will be addressed later on for RevisionRow.tsx

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionRow.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionRow.tsx:31]: I added this suggestion in the clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488) and it will be addressed later on for RevisionRow

File: src/components/CompareResults/SubtestsResults/SubtestsRevisionRow.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsRevisionRow.tsx:219]: I added this suggestion in the clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488) and it will be addressed later on for both components.

File: src/components/CompareResults/TableHeader.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/TableHeader.tsx:135]: I added this suggestion in the clean-up [subtask](https://mozilla-hub.atlassian.net/browse/PCF-488) and it will be addressed later on

File: src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx
Comment (julienw) [inline src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx:150]: Can you please file a JIRA ticket to merge this with the other results table? I think we can extract a common component doing the filtering... but I don't want to hold up this PR for this.

**Review by julienw — COMMENTED** [review]
Thanks!

I'm happy to land this once you change the width of the subtest name. I'm also happy if you want to do it in a quick follow-up instead.

But please also revert some of the changes I highlight in comments.

Please don't forget to run prettier before landing either.

**Review by julienw — APPROVED** [review]

File: src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx:150]: https://mozilla-hub.atlassian.net/browse/PCF-489

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (beatrice-acasandrei) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:58]: Removing Await will results in `TypeError: results is undefined`

File: src/components/CompareResults/TableHeader.tsx
Comment (beatrice-acasandrei) [inline src/components/CompareResults/TableHeader.tsx:121]: @julienw is this what you had in mind regarding the gridWidth?

File: src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx
Comment (julienw) [reply src/components/CompareResults/SubtestsResults/SubtestsResultsMain.tsx:58]: That's true, but that's not the right fix.
The error happens because you're using `useAsyncValue` in `SubtestsResultsTable`. `useAsyncValue` works with `Await`.
But `results` is not a promise! so there's no need to use `Await` and there's no need to use `useAsyncvalue`, you could use `useLoaderData` in `SubtestsResultsTable` or directly pass down `results` using a prop.

**Pull request merged**
